File: frontend/src/types/metrics.ts
export interface FunnelStep {
  label: string;
  count: number;
  color: string;
}

export interface DayCount {
  dia: string;
  multiplas: number;
  unica: number;
}

export interface TipoCount {
  tipo: string;
  count: number;
  color: string;
}

export interface DashboardMetrics {
  processos_hoje: number;
  leads_capturados: number;
  audiencias_encontradas: number;
  valor_total: number;
  funnel: FunnelStep[];
  audiencias_por_dia: DayCount[];
  tipos_audiencia: TipoCount[];
}

File: backend/app/models/metrics.py
from pydantic import BaseModel
from typing import List


class FunnelStep(BaseModel):
    label: str
    count: int
    color: str


class DayCount(BaseModel):
    dia: str
    multiplas: int
    unica: int


class TipoCount(BaseModel):
    tipo: str
    count: int
    color: str


class DashboardMetrics(BaseModel):
    processos_hoje: int
    leads_capturados: int
    audiencias_encontradas: int
    valor_total: float
    funnel: List[FunnelStep]
    audiencias_por_dia: List[DayCount]
    tipos_audiencia: List[TipoCount]
